File: client_jfx/src/main/java/model/game/elements/Block.java
package model.game.elements;

import javafx.scene.image.Image;

public class Block extends LevelElement {
    private int lives;

    public Block(Image image, LevelElementType type, double positionX, double positionY, double width, double height, double rotation) {
        super(image, type, positionX, positionY, width, height, rotation);
        if(type == LevelElementType.BLOCK_METAL){
            lives = 10000;
        } else if(type == LevelElementType.BLOCK_STONE){
            lives = 2;
        } else if(type == LevelElementType.BLOCK_WOOD){
            lives = 3;
        }
    }

    public Block(Image image, LevelElementType type, double positionX, double positionY, double width, double height) {
        super(image, type, positionX, positionY, width, height);
        if(type == LevelElementType.BLOCK_METAL){
            lives = 10000;
        } else if(type == LevelElementType.BLOCK_STONE){
            lives = 2;
        } else if(type == LevelElementType.BLOCK_WOOD){
            lives = 3;
        }
    }

    public void setLives(int lives) {
        this.lives = lives;
    }

    public int getLives() {
        return lives;
    }
}

File: client_python/src/model/game/elements/Block.py
from model.game.elements.LevelElementType import LevelElementType
from model.game.elements.LevelElement import LevelElement


class Block(LevelElement):
    def __init__(self, elementType, imagePath, positionX, positionY, width, height):
        super().__init__(elementType, imagePath, positionX, positionY, width, height, 360)

        if elementType == LevelElementType.BLOCK_METAL:
            self.lives = 10001
        elif elementType == LevelElementType.BLOCK_STONE:
            self.lives = 3
        elif elementType == LevelElementType.BLOCK_WOOD:
            self.lives = 1

